src/main/java/user/createcorporate/CorporateGenerateController.java

```diff
@@ -0,0 +1,40 @@
+package user.createcorporate;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class CorporateGenerateController {
+    final ICorporateGenerateInput input;
+
+    /**
+     *
+     * @param input is passed to the controller via input boundary
+     */
+
+    @Autowired
+    public CorporateGenerateController(ICorporateGenerateInput input) {
+        this.input = input;
+    }
+
+    /**
+     *
+     * @param requestModel is passed in to the responseModel creation process
+     * @return interactor is returned
+     */
+
+    @PostMapping("/users/createCorporate")
+    public ResponseEntity<CorporateGenerateResponseModel> create(CorporateGenerateRequestModel requestModel){
```

Comment:
Let the controller take in an extra parameter `Principal principal`, which contains the authenticated user

If `principal` is null, there is no authenticated user so we can reject request right away
If `principal.getName()` does not match `userId`, then the user could be faking a request so we need to reject that as well